components/runtime/squeak5/Makefile

```diff
@@ -0,0 +1,100 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Package Makefile for the "OpenSmalltalk Cog Spur" Squeak Smalltalk V5 system
+# See http://squeak.org and http://opensmalltalk.org
+#
+
+# opensmalltalk-vm can be built both in 32 and 64bit
+# but generally SqueakV5 images are 64bit
+BUILD_BITS=64
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		squeak-5
+COMPONENT_VERSION=	5.0.7
+GIT_TAG=		sun-v$(COMPONENT_VERSION)
+PLUGIN_REV=		5.0-202005021055
+COMPONENT_SUMMARY=	The Squeak V5 Smalltalk Virtual Machine
+COMPONENT_PROJECT_URL=	http://www.squeak.org
+COMPONENT_FMRI=		runtime/squeak-5
+COMPONENT_CLASSIFICATION=	Development/Other Languages
+
+# See http://wiki.squeak.org/squeak/933
+# See http://wiki.squeak.org/squeak/159
+COMPONENT_LICENSE=	MIT
+COMPONENT_LICENSE_FILE=	squeak5.license
+
+COMPONENT_SRC=		opensmalltalk-vm-$(GIT_TAG)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:50c180b174048acd98a7d9705f0a40a3dc8bdc62b42d88eb7e20e71834184e4c
+COMPONENT_ARCHIVE_URL=	https://codeload.github.com/cstes/opensmalltalk-vm/tar.gz/$(GIT_TAG)
+
+TEST_TARGET= $(NO_TESTS)
+
+include $(WS_MAKE_RULES)/common.mk
+
+PKG_OPTIONS += -D PLUGIN_REV="$(PLUGIN_REV)"
+
+PATH=$(PATH.gnu)
+
+# opensmalltalk configure script checks for plugins files in builddir
+COMPONENT_PRE_CONFIGURE_ACTION = ( cp plugins.ext plugins.int $(BUILD_DIR_64) )
+
+# the default CFLAGS.gcc will be -m64 -O3 but this does not work for us
+# opensmalltalk code is not compatible with the gcc optimizer -O2 or higher
+# also the -g (debug) flag is intentional, perhaps omit-frame-pointer issues
+CFLAGS=			-g -DNDEBUG -DDEBUGVM=0
+
+CONFIGURE_SCRIPT= $(SOURCE_DIR)/platforms/unix/config/configure
+CONFIGURE_OPTIONS +=	--with-vmversion=5.0
+CONFIGURE_OPTIONS +=	--disable-dynamicopenssl
+CONFIGURE_OPTIONS +=	--without-libtls
+CONFIGURE_OPTIONS +=	--with-src=spurstack64src
+CONFIGURE_OPTIONS +=	--disable-cogit
+CONFIGURE_OPTIONS +=	--without-vm-display-fbdev
+CONFIGURE_OPTIONS +=	--without-npsqueak
+
+CONFIGURE_ENV +=	TARGET_ARCH="-m64"
+
+# the opensmalltalk Makefile does not use DESTDIR
+COMPONENT_INSTALL_ARGS=  	ROOT=$(PROTO_DIR)
+COMPONENT_INSTALL_TARGETS= 	install-squeak install-doc install-plugins
+
+# opensmalltalk-vm installs a wrapper script squeak in /usr/bin
+# rename those scripts and the manpage to squeak5
+# to avoid conflict with the squeak-4 package (mediator "squeak')
+# also their old Makefile installs in /usr/doc (for SunOS the location)
+# move that to $(PROTOUSRSHAREDOCDIR) /usr/share/doc as it is nowadays
+COMPONENT_POST_INSTALL_ACTION = ( \
+  $(CP) $(PROTOUSRBINDIR)/squeak $(PROTOUSRBINDIR)/squeak5; \
```

Comment:
Likely we want $(MV) here to avoid confusing next person, looking on sample-manifest

Reply:
@pyhalov In such cases I find it easier to address renaming directly in the manifest, like:
`file usr/bin/squeak path=usr/bin/squeak4`